eq3bt/bleakconnection.py

```diff
@@ -0,0 +1,121 @@
+"""
+Bleak connection backend.
+This creates a new event loop that is used to integrate bleak's
+asyncio functions to synchronous architecture of python-eq3bt.
+"""
+import asyncio
+import codecs
+import contextlib
+import logging
+from typing import Optional
+
+from bleak import BleakClient, BleakError
+
+from . import BackendException
+
+DEFAULT_TIMEOUT = 1
+
+# bleak backends are very loud, this reduces the log spam when using --debug
+logging.getLogger("bleak.backends").setLevel(logging.WARNING)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class BleakConnection:
+    """Representation of a BTLE Connection."""
+
+    def __init__(self, mac, iface):
+        """Initialize the connection."""
+
+        self._conn: Optional[BleakClient] = None
+        self._mac = mac
+        self._iface = iface
+        self._callbacks = {}
+        self._notifyevent = asyncio.Event()
+        self._notification_handle = None
+        self._loop = asyncio.new_event_loop()
+
+    def __enter__(self):
+        """
+        Context manager __enter__ for connecting the device
+        :rtype: BTLEConnection
+        :return:
+        """
+        _LOGGER.debug("Trying to connect to %s", self._mac)
+
+        # TODO: bleak doesn't support interface binding?
```

Comment:
I think it does 😄 
https://github.com/hbldh/bleak/issues/76

Reply:
Oh, looks like it's indeed possible (at least on the bluez backend)! It doesn't like if `adapter` is passed as `None` which is what happens if no interface is defined, so I'll have to see if this is better to be patched at bleak's or at this lib's end, but I'll figure it out.